pkg/kubectl/cmd/util/factory_object_mapping.go

```diff
@@ -151,15 +156,22 @@ func (f *ring1Factory) UnstructuredClientForMapping(mapping *meta.RESTMapping) (
 	return restclient.RESTClientFor(cfg)
 }
 
-func (f *ring1Factory) Describer(mapping *meta.RESTMapping) (kubectl.Describer, error) {
+func (f *ring1Factory) PrinterForCommand(cmd *cobra.Command) (printers.ResourcePrinter, bool, error) {
+	mapper, typer := f.Object()
```

Comment:
No peer references.  This will break downstream in openshift and anyone else trying to use it.  You'll need to move this to ring2.